core/jobs/job_test_utils_test.py

```diff
@@ -93,6 +93,11 @@ def tearDown(self) -> None:
     def test_run_job(self) -> None:
         self.run_job()
 
+        # Here we use MyPy ignore because the method 'run' is of
```

Comment:
But if this is the case, how do we know that assert_called() is actually a method on self.job.run (i.e. that this code will actually work)?

Reply:
That's a runtime thing, right? and MyPy deals only with typing, and here run is of callable type and callable type does not contain this `assert_called` attribute.

Reply:
Sorry, I'm confused. Where is assert_called() defined? How do you know this is a runtime thing .. what's the evidence?

Reply:
I tried to find where this `assert_called()` is defined but was unable to find it.

The main thing here is that `run_job` is of callable type and currently MyPy does not support the addition of attributes in Callable types.

I found an issue that was already opened in MyPy, basically, they are currently in discussion about this feature and this feature is not implemented yet. -- issue https://github.com/python/mypy/issues/2087

But yeah, at the last of the conversation someone suggested a hack for this issue but that seems odd to me. -- suggested hack/way to suppress the error https://github.com/python/mypy/issues/2087#issuecomment-769266912

Reply:
Hm, this is indeed a bit odd.  I dug into it using `git blame` but I also can't find any information on what this is.

For this and the cases below, I suggest writing a TODO to investigate where assert_called() and return_value come from, and whether they can be replaced by something more standard. (Give more info in the issue than this to explain the context properly.) I think this is something that does needs to be fixed in the longer term, but not urgently enough to block this PR on. Does that sound good?

/cc @vojtechjelinek

Reply:
TODO is added